feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -154,6 +261,15 @@ class PreviewViewModel @AssistedInject constructor(
     fun setCaptureMode(captureMode: CaptureMode) {
         viewModelScope.launch {
             cameraUseCase.setCaptureMode(captureMode)
+        }.also {
+            _previewUiState.update { old ->
+                (old as? PreviewUiState.Ready)?.copy(
+                    captureModeToggleUiState = getCaptureToggleUiState(
+                        old.systemConstraints,
+                        cameraUseCase.getCurrentSettings().value!!
+                    )
+                ) ?: old
+            }
```

Comment:
If user updates dynamicRange, imageOutputFormat, or CaptureMode, currentSettings is updated by CameraXCameraUseCase, but CaptureToggleUiState isn't. But changing any of the three could result in CaptureToggleUiState changing, therefore shouldn't we also update it here?

Reply:
Otherwise CaptureToggleUiState is only updated in the init block

Reply:
The init block is basically a `while` loop (with suspending wait). It listens for all changes from `CameraXCameraUseCase` and will update `CaptureToggleUiState` whenever any of the other settings change.

We probably want to restructure the init block a little bit since it also recalculates `CaptureToggleUiState` whenever the zoom level changes, but that can happen in a different PR.